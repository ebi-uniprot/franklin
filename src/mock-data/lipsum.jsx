import { loremIpsum } from 'lorem-ipsum';
import { v1 } from 'uuid';
import { fillArray, capitaliseFirstLetter } from '../utils';

export const getLipsumWords = () =>
  capitaliseFirstLetter(loremIpsum({ count: 2, units: 'words' }));

export const getLipsumSentences = () =>
  loremIpsum({
    sentenceLowerBound: 2,
    sentenceUpperBound: 30,
  });

export const getLipsumObjectArray = ({
  numberElements,
  keys,
  idKey = 'id',
  type = 'sentences',
}) =>
  fillArray(numberElements, () => {
    const dataPoint = { [idKey]: v1() };
    keys.forEach((key) => {
      const text =
        type === 'sentences' ? getLipsumSentences() : getLipsumWords();
      dataPoint[key] = text;
    });
    return dataPoint;
  });

export default getLipsumObjectArray;
