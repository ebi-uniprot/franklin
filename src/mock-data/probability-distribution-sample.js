import random from 'random';

export const getGaussianSample = (mu, sigma, n, min = null, max = null) => {
  const normal = Array.from({ length: n }, random.normal(mu, sigma));
  // If we want to bound the sampling
  if (min || max) {
    return normal.map((number) => {
      if (min !== null && number < min) {
        return min;
      }
      if (max !== null && number > max) {
        return max;
      }
      return number;
    });
  }
  return normal;
};

export const getUniformSample = (min, max, n) =>
  Array.from({ length: n }, random.uniform(min, max));
