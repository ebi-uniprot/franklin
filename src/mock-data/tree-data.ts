import {
  getFlattenedPaths,
  restructureFlattenedTreeDataForAutocomplete,
} from '../utils';

export const treeData = [
  {
    label: 'Item 1',
    id: 'item_1',
    items: [
      {
        label: 'Item 1a',
        id: 'item_1a',
      },
      {
        label: 'Item 1b',
        id: 'item_1b',
        items: [
          {
            label: 'Item 1b A',
            id: 'item_1b_A',
          },
          {
            label: 'Item 1b B',
            id: 'item_1b_B',
          },
        ],
      },
    ],
  },
  {
    label: 'Some Item 2',
    id: 'item_2',
  },
];

const flatPaths = getFlattenedPaths(treeData);
export const flattenedPaths =
  restructureFlattenedTreeDataForAutocomplete(flatPaths);
