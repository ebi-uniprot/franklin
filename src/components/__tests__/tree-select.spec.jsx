import { render } from '@testing-library/react';

import TreeSelect from '../tree-select';

import { treeData } from '../../mock-data/tree-data';

describe('TreeSelect component', () => {
  test('should render', () => {
    const { asFragment } = render(
      <TreeSelect data={treeData} onSelect={(d) => d} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('TreeSelect component with filter', () => {
  test('should render', () => {
    const { asFragment } = render(
      <TreeSelect data={treeData} onSelect={(d) => d} autocomplete />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
