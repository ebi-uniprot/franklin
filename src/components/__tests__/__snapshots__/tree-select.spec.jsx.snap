// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeSelect component should render 1`] = `
<DocumentFragment>
  <div
    class="dropdown-container"
  >
    <button
      class="button primary dropdown"
      type="button"
    >
      Select
    </button>
    <div
      class="dropdown-menu"
    />
  </div>
</DocumentFragment>
`;

exports[`TreeSelect component with filter should render 1`] = `
<DocumentFragment>
  <div
    class="dropdown-container"
  >
    <button
      class="button primary dropdown"
      type="button"
    >
      Select
    </button>
    <div
      class="dropdown-menu"
    />
  </div>
</DocumentFragment>
`;
